NDTensors/test/test_dense.jl

```diff
@@ -232,6 +232,22 @@ using .NDTensorsTestUtils: devices_list
   end
 
   # Only CPU backend testing
+  @testset "Contract with exotic types" begin
+    # BigFloat is not supported on GPU
+    M = Tensor(BigFloat, (1, 2, 3))
+    fill!(M, BigFloat(randn(Float64)))
+    O = Tensor(M, (2, 3))
+    N = Tensor([BigFloat(1.0)], (1,))
+  
+    @test O ≈ contract(M, (-1, 2, 3), N, (-1,))
+    O = similar(M)
+    ## TODO find a way to test the next line, its not possible to access an undefined
+    ## reference
```

Comment:
I don't understand this comment.

Reply:
From what I can tell, it doesn't make sense to test `!iszero(β)` and `O` initialized with `similar`, since that wouldn't give a well defined result (whether you are using `BigFloat` or a simpler element type). If you want to test that case it seems like you should initialize the data, for example with `randn!`.

Reply:
Sorry this was a comment for myself. I replaced the test with this
```
tp = similar(t)
NDTensors.contract!(tp, (1, 2, 3), t, (-1, 2, 3), v, (-1,), 0.0, 1.0)
@test inds(tp) == inds(t)
    ``` 
    But its not possible to do a test like `@test tp == t` because accessing elements of the array throws an UndefRefError
    ```
    
Error During Test at /Users/kpierce/.julia/dev/ITensors/NDTensors/test/test_dense.jl:247
  Test threw exception
  Expression: m == t
  UndefRefError: access to undefined reference
  ```

Reply:
My point was that I don't think we should expect `contract!` to work when `!iszero(β)` and `tp` has undefined data (for example initialized with `similar`) so that doesn't make sense to test anyway.

Reply:
Makes sense, I removed the test. Thanks!

Reply:
Calling `contract!` in that way just isn't well defined and users shouldn't do that, just like users shouldn't use call `LinearAlgebar.mul!(c, a, b, α, β)` if `c` has undefined data and `!iszero(β)`. It may or may not error depending on the element type, but either way it isn't a good idea to do that since the answer would be undefined (you could just get complete junk data as the result of `c`).

Even if `iszero(α)`, the operation you are doing is `c .= β .* c`, so if `c` is filled with junk (undefined data) then the answer you get would be junk.